### PR TITLE
Moved harmonization logic to Term classes, added MIDS 1 terms

### PR DESCRIPTION
**Pull request by samleeflang**
Added terms Modified, License and ObjectType

### CODE REVIEW

File: src/test/java/eu/dissco/core/translator/terms/specimen/SpecimenNameTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    var specimenNameString = "SpecimenName";
+    var archiveField = new ArchiveField(0, DwcTerm.scientificName);
+    given(archiveFile.getField("dwc:scientificName")).willReturn(archiveField);
+    given(rec.value(archiveField.getTerm())).willReturn(specimenNameString);
+
+    // When
+    var result = specimenName.retrieveFromDWCA(archiveFile, rec);
+
+    // Then
+    assertThat(result).isEqualTo(specimenNameString);
```
Comment (TomDijkema) [inline src/test/java/eu/dissco/core/translator/terms/specimen/SpecimenNameTest.java:38]: Why is the specimenNameString a static string?

Comment (TomDijkema) [reply src/test/java/eu/dissco/core/translator/terms/specimen/SpecimenNameTest.java:38]: Oh wait, is it because it is a test

**Review by southeo — APPROVED** [review]
Looks good to me. To clarify: is the plan then to have an explicit Term class for each term we use in OpenDS?

File: src/test/java/eu/dissco/core/translator/terms/specimen/SpecimenNameTest.java (hunk 1 above)
Comment (samleeflang) [reply src/test/java/eu/dissco/core/translator/terms/specimen/SpecimenNameTest.java:38]: Yes could be any string, all that matters is that it comes out equal

File: src/main/java/eu/dissco/core/translator/domain/DigitalMediaObject.java [hunk 2]
```diff
@@ -5,11 +5,8 @@
...
     String physicalSpecimenId,
-    String mediaUrl,
-    String format,
-    String sourceSystemId,
-    JsonNode data,
-    JsonNode originalData,
-    String idType) {
+    String idType,
+    JsonNode attributes,
+    JsonNode originalAttributes) {
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/domain/DigitalMediaObject.java:19]: They have been mapped to a harmonized field and are included in `attributes`.

**Comment by samleeflang** [comment]
> Looks good to me. To clarify: is the plan then to have an explicit Term class for each term we use in OpenDS?

Yes for each harmonized term we will create a Term class with all the logic that is needed for harmonization.
Default behaviour will be to check the mapping for a default value or an explicit field mapping (termMapper). If those are not present we can define fields to check and extended logic in the particular Term. Hopefully this provides both the generic approach and gives room for complex logic (see MediaType). It reduced the size of the services (DwcaSerivce/BiocaseService) and their complexity. It also provides a simple overview in case of bugs or logic extensions as you can just look up the Term class and see all its logic. It was a bit inspired by the Command Pattern

File: src/main/java/eu/dissco/core/translator/service/DwcaService.java [hunk 3]
```diff
@@ -116,21 +132,28 @@ private void extractACMultimedia(ArchiveFile extension) throws JsonProcessingExc
       if (idTobeIgnored.contains(rec.id())) {
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/service/DwcaService.java:134]: Well spotted :+1:

File: src/main/java/eu/dissco/core/translator/Profiles.java [hunk 4]
```diff
@@ -1,7 +1,7 @@
 package eu.dissco.core.translator;
 
 public class Profiles {
-  public static final String BIO_CASE = "bioCase";
+  public static final String BIOCASE = "biocase";
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/core/translator/Profiles.java:4]: Officially it is called: BioCASe, but lowercase should be better for code :)

File: src/test/java/eu/dissco/core/translator/terms/specimen/ModifiedTest.java [hunk 5]
```diff
@@ -41,16 +40,15 @@ void testRetrieveFromDWCA() {
   @Test
   void testRetrieveFromABCD() {
     // Given
-    long epoch = 1674553668909L;
-    var instant = Instant.ofEpochMilli(epoch);
+    String modifiedString = "1674553668909";
     var unit = new ObjectMapper().createObjectNode();
-    unit.put("abcd:dateLastEdited", epoch);
+    unit.put("abcd:dateLastEdited", modifiedString);
```
Comment (TomDijkema) [inline src/test/java/eu/dissco/core/translator/terms/specimen/ModifiedTest.java:45]: Did the ABCD extensions came to discussion yet? (EFG / DNA)

File: src/main/java/eu/dissco/core/translator/service/BioCaseService.java [hunk 6]
```diff
@@ -171,76 +180,81 @@ private void mapABCD206(XMLEventReader xmlEventReader)
...
-            webClientProperties.getSourceSystemId(),
-            removeMappedFields(unitData),
-            unitData,
-            null
+            termMapper.retrieveFromABCD(new Type(), unitAttributes),
+            harmonizeAttributes(datasets, unitAttributes, physicalSpecimenIdType, organizationId),
+            unitAttributes
         );
+
         log.debug("Result digital Specimen: {}", digitalSpecimen);
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/core/translator/service/BioCaseService.java:199]: Do we want to make a kind of guideline for creating logs within the DiSSCo environment? To apply to the other back- and front-end applications (UCAS, ELViS, etc.)

File: src/main/java/eu/dissco/core/translator/service/DwcaService.java [hunk 7]
```diff
@@ -116,21 +132,28 @@ private void extractACMultimedia(ArchiveFile extension) throws JsonProcessingExc
...
             type,
             rec.id(),
-            rec.value(extension.getField("ac:accessUri").getTerm()),
-            rec.value(extension.getField(DCTERMS_FORMAT).getTerm()),
-            webClientProperties.getSourceSystemId(), getData(extension, rec, false),
-            getData(extension, rec, true),
-            "DWCA");
+            "DWCA",
+            harmonizeMedia(extension, rec),
+            getOriginalAttributes(extension, rec));
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/core/translator/service/DwcaService.java:144]: I would move the ending of such a statement (the ); part) to one line below to complete the indent with the top

File: src/test/java/eu/dissco/core/translator/terms/specimen/ModifiedTest.java (hunk 5 above)
Comment (TomDijkema) [reply src/test/java/eu/dissco/core/translator/terms/specimen/ModifiedTest.java:45]: Nevermind, saw the modules for efg

File: src/main/java/eu/dissco/core/translator/Profiles.java (hunk 4 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/Profiles.java:4]: Yeah that's why... did this wrong a couple of times and then thought, just keep it simple

File: src/main/java/eu/dissco/core/translator/service/BioCaseService.java (hunk 6 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/service/BioCaseService.java:199]: Not sure, never had a guideline. I think the general rule is do logging, enough to trace what is going on but not so much you can't find anything

Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/service/BioCaseService.java:199]: Created this ticket https://naturalis.atlassian.net/browse/DD-190

**Pull request merged**